### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by BenM4k**
In this update, we added unit tests for all the methods in all the classes:
- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- All classes that wecreated while implementing the Decorator design pattern

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    expect(book.title).to eq('How to make one million in 1 hour')
+    expect(book.author).to eq('Mr Money')
+  end
+
+  it 'can add rentals' do
+    book = Book.new('How to make one million in 1 hour', 'Mr Money')
+    rental = double('Rental')
+    book.add_rental(rental)
+    expect(book.rentals).to include(rental)
+  end
```
Comment (MahmoudBakr23) [inline spec/book_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    classroom = Classroom.new('Class Five')
+    expect(classroom.label).to eq('Class Five')
+  end
+
+  it 'can add student' do
+    classroom = Classroom.new('Class two')
+    student = double('Student')
+    classroom.add_student(student)
+    expect(classroom.students).to include(student)
+  end
```
Comment (MahmoudBakr23) [inline spec/classroom_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

File: spec/decorator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+end
+
+describe TrimmerDecorator do
+  person = Person.new(1, 'maximilianus', 25, true)
+  capitalized_person = CapitalizeDecorator.new(person)
+  trimmed_person = TrimmerDecorator.new(capitalized_person)
+  it '' do
+    expect(trimmed_person.correct_name).to eq('Maximilian')
+  end
+end
```
Comment (MahmoudBakr23) [inline spec/decorator_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍 
- [ ] Please make sure to make the tests of each class in separate _spec file for a cleaner code base and easier-to-track. 👍

File: spec/person_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    expect(person.rentals).to include(rental)
+  end
+
+  it 'correctly checks if the person can use services' do
+    expect(person.can_use_services?).to be(true)
+
+    person.age = 15
+    person.parent_permission = false
+    expect(person.can_use_services?).to be(false)
+  end
```
Comment (MahmoudBakr23) [inline spec/person_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

File: spec/rental_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require_relative '../rental'
+describe Rental do
+  it 'has an book, person, and date' do
+    person = Person.new(1, 'John Dada', 30, true)
+    book = Book.new('How to make one million in 1 hour', 'Mr Money')
+    rental = Rental.new(book, person, '2022-01-01')
+    expect(rental.book).to eq(book)
+    expect(rental.person).to eq(person)
+  end
```
Comment (MahmoudBakr23) [inline spec/rental_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

File: spec/student_spec.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require_relative '../student'
+
+describe Student do
+  it 'should have the same properties' do
+    student = Student.new(1, 17, 'Myself', false)
+    expect(student.id).to eq(1)
+    expect(student.name).to eq('Myself')
+  end
```
Comment (MahmoudBakr23) [inline spec/student_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

File: spec/teacher_spec.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+require_relative '../teacher'
+
+describe Teacher do
+  it 'should have the same properties' do
+    teacher = Teacher.new(1, 'Time', 'Myself', 35)
+    expect(teacher.id).to eq(1)
+    expect(teacher.name).to eq('Myself')
+  end
```
Comment (MahmoudBakr23) [inline spec/teacher_spec.rb]: - [ ] Please refactor the test file to use `describe` and `context` for each test block to identify it and run it separately. 🙏 This ensures the readability of the tests and also gets the best results out of them. 👍

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello Team 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Nicely done setting Rspec
- Well done testing all the methods of each class
- Descriptive README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider the following:
- In pair-programming projects, please keep in mind to have equally distributed commit messages in the PR. I know the work was divided equally. However, after every change, push the changes with a commit message. This will ensure that all members are listed properly and the code is easier to get reviewed. 🙏 

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @BenM4k and @yemidada,

## Highlights!
- Good code organization and working tests✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**